File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = [0]
    def dfs(index,number,target,answer):
        if index==len(numbers)-1:
            if number==target:
                answer[0]+=1
            return
        dfs(index+1,number+numbers[index+1],target,answer)
        dfs(index+1,number-numbers[index+1],target,answer)
    dfs(-1,0,target,answer)
    return answer[0]

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;
int answer = 0;
void dfs(vector<int> &numbers,int target,int depth,int num){
    if (depth==numbers.size()){
        if (num==target) answer++;
        return;
    }
    
    dfs(numbers,target,depth+1,num-numbers.at(depth));
    dfs(numbers,target,depth+1,num+numbers.at(depth));
}

int solution(vector<int> numbers, int target) {
    dfs(numbers,target,0,0);
    return answer;
}